jwst/outlier_detection/outlier_detection_step.py

```diff
@@ -96,11 +95,18 @@ def _build_reffile_container(self, reftype):
         reffile_to_model = {'gain': datamodels.GainModel,
             'readnoise': datamodels.ReadnoiseModel}
 
-        reffiles = [self.get_reference_file(im, reftype) for im in self.input_models]
+        if reftype in self.input_models[0].meta.ref_file._instance:
+            reffile_names = [tm.meta.ref_file._instance[reftype]['name'] for tm in self.input_models]
+            if len(set(reffile_names)) == 1:
+               reffiles = [self.get_reference_file(self.input_models[0],reftype)]*len(reffile_names)
+            else:
+                reffiles = [self.get_reference_file(im, reftype) for im in self.input_models]                 
```

Comment:
I believe @jaytmiller  has implemented a function which expands the name saved in `meta.ref_file.name` . Perhaps he or @stscirij can confirm this.